superset/utils/core.py

```diff
@@ -902,6 +902,8 @@ def send_email_smtp(  # pylint: disable=invalid-name,too-many-arguments,too-many
     msg_mutator = config["EMAIL_HEADER_MUTATOR"]
     # the base notification returns the message without any editing.
     new_msg = msg_mutator(msg, **(header_data or {}))
+    if new_msg["To"].split(", ") != recipients:
```

Comment:
I'm not certain whether this should be handled here for the specific bespoke use case or we should change the `EMAIL_HEADER_MUTATOR` (or similar) to return `recipients` as well.

Reply:
Hey @john-bodley, thanks for taking a look. Ideally we'd love to have control over the actual email object including the `dry_run` flag but I thought this would be a simple step in that direction that would suit our immediate needs. 

Changing the return values for this would probably cause a bit of a headache for many existing users. How would you suggest we continue?